tgui/packages/tgui/interfaces/ExodroneConsole.tsx

```diff
@@ -0,0 +1,807 @@
+  deep_scan_complete: boolean,
+  events: Array<ExplorationEventData>
+}
+
+
+enum DroneStatusEnum {
+  Idle = "idle",
+  Travel = "travel",
+  Exploration = "exploration",
+  Adventure = "adventure",
+  Busy = "busy"
+}
+
+enum CargoType {
+  Tool = "tool",
+  Cargo = "cargo",
+  Empty = "empty"
+}
+
+type CargoData = {
+  type: CargoType,
+  name: string
+}
+
+type DroneBasicData = {
+  name: string,
+  description: string,
+  controlled: boolean,
+  ref: string,
+}
+
+type ExodroneConsoleData = {
+  signal_lost: boolean,
+  drone: boolean,
+  all_drones?: Array<DroneBasicData>
+  drone_status?: DroneStatusEnum,
+  drone_name?: string,
+  drone_integrity?: number,
+  drone_max_integrity?: number,
+  drone_travel_coefficent?: number,
+  drone_log?: Array<string>,
+  configurable?: boolean,
+  cargo?: Array<CargoData>,
+  can_travel?: boolean,
+  travel_error: string,
+  sites?: Array<SiteData>,
+  site?: SiteData,
+  travel_time?: number,
+  travel_time_left?: number,
+  wait_time_left?: number,
+  wait_message?: string,
+  event?: FullEventData,
+  adventure_data?: AdventureData,
+  // ui_static_data
+  all_tools: Record<string, ToolData>,
+  all_bands: Record<string, string>
+}
+
+type ToolData = {
+  description: string,
+  icon: string
+}
+
+export const ExodroneConsole = (props, context) => {
+  const { data } = useBackend<ExodroneConsoleData>(context);
+  const {
+    signal_lost,
+  } = data;
+
+  const [
+    choosingTools,
+    setChoosingTools,
+  ] = useLocalState(context, 'choosingTools', false);
+
+  return (
+    <Window width={650} height={500}>
+      {!!signal_lost && <SignalLostModal />}
+      {!!choosingTools && <ToolSelectionModal />}
+      <Window.Content>
+        <ExodroneConsoleContent />
+      </Window.Content>
+    </Window>
+  );
+};
+
+const SignalLostModal = (props, context) => {
+  const { act } = useBackend(context);
+  return (
+    <Modal backgroundColor="red" textAlign="center" width={30} height={22} p={0} style={{ "border-radius": "5%" }}>
+      <img src={nt_logo} width={64} height={64} />
+      <Box backgroundColor="black" textColor="red" fontSize={2} style={{ "border-radius": "-10%" }}>CONNECTION LOST</Box>
+      <Box p={2} italic>
+        Connection to exploration drone interrupted.
+        Please contact nearest Nanotrasen Exploration Division
+        representative for further instructions.
+      </Box>
+      <Icon name="exclamation-triangle" textColor="black" size={5} />
+      <Box><Button color="danger" style={{ "border": "1px solid black" }} onClick={() => act("confirm_signal_lost")}>Confirm</Button></Box>
+    </Modal>);
+};
+
+const DroneSelectionSection = (props, context) => {
+  const { act, data } = useBackend<ExodroneConsoleData>(context);
+  const {
+    all_drones,
+  } = data;
+
+  return (
+    <Section scrollable fill title="Exploration Drone Listing">
+      <Stack vertical>
+        {all_drones.map(drone => (
+          <>
+            <Stack.Item grow key={drone.ref}>
```

Comment:
That's a key used in the wrong place, it should be on the fragment, but the only way to specify key on the fragment is to import it from `inferno`, and use it the old way:
```js
<Fragment key={drone.ref}>
```